File: mavenproject1/src/main/java/Objects/World.java

package Objects;

import java.util.ArrayList;
import java.util.List;



public class World {
    
    private String name = null;
    private int rows;
    private int cols;
    private Config config;
    private List<Piece> board;
    private List<Box> boxes;
    private List<Target> targets;
    private Player player;
    
    
    public World() {
        
        board = new ArrayList<Piece>();
        boxes = new ArrayList<Box>();
        targets = new ArrayList<Target>();
        player = null;
        config = null;
        
    }
    public void createWorld(String name, int rows, int cols){
        this.name = name;
        this.rows = rows;
        this.cols = cols;
    }
    public void createPlayer(Player player){
        this.player = player;
        
    }
    
    public void createBox(Box box){
        boxes.add(box);
    }
    
    public void createTarget(Target target ){
        targets.add(target);
    }
    
    public void createPiece(Piece piece){
        board.add(piece);
    }
    
    public Config getConfig() {
        return config;
    }

    public void setConfig(Config config) {
        this.config = config;
    }

    public List<Piece> getBoard() {
        return board;
    }

    public void setBoard(List<Piece> Board) {
        this.board = Board;
    }

    public List<Box> getBoxes() {
        return boxes;
    }

    public void setBoxes(List<Box> Boxes) {
        this.boxes = Boxes;
    }

    public List<Target> getTargets() {
        return targets;
    }

    public void setTargets(List<Target> targets) {
        this.targets = targets;
    }

    public Player getPlayer() {
        return player;
    }

    public void setPlayer(Player player) {
        this.player = player;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getRows() {
        return rows;
    }

    public void setRows(int rows) {
        this.rows = rows;
    }

    public int getCols() {
        return cols;
    }

    public void setCols(int cols) {
        this.cols = cols;
    }
    
    
    
}

File: app/src/main/java/com/example/proyecto1compi1_2023/objects/World.kt
package com.example.proyecto1compi1_2023.objects

class World  : java.io.Serializable{
    var name: String? = null
    var rows = 0
    var cols = 0
    var config: Config?
    private var board: MutableList<Piece>
    private var boxes: MutableList<Box>
    private var targets: MutableList<Target>
    var player: Player?

    init {
        board = ArrayList()
        boxes = ArrayList()
        targets = ArrayList()
        player = null
        config = null
    }

    fun createWorld(name: String?, rows: Int, cols: Int) {
        this.name = name
        this.rows = rows
        this.cols = cols
    }

    fun createPlayer(player: Player?) {
        this.player = player
    }

    fun createBox(box: Box) {
        boxes.add(box)
    }

    fun createTarget(target: Target) {
        targets.add(target)
    }

    fun createPiece(piece: Piece) {
        board.add(piece)
    }

    fun getBoard(): List<Piece> {
        return board
    }

    fun setBoard(Board: MutableList<Piece>) {
        board = Board
    }

    fun getBoxes(): List<Box> {
        return boxes
    }

    fun setBoxes(Boxes: MutableList<Box>) {
        boxes = Boxes
    }

    fun getTargets(): List<Target> {
        return targets
    }

    fun setTargets(targets: MutableList<Target>) {
        this.targets = targets
    }
}